libs/wingc/src/jsify.rs

```diff
@@ -758,9 +805,88 @@ impl<'a> JSifier<'a> {
+				// To allow for struct validation at runtime this will generate a JS class that has a static
+				// validate method that will check the type of each field and return an array of errors if any
+
+				let mut code = CodeMaker::default();
+				code.open(format!("module.exports = function(fromInline) {{"));
+				code.open(format!("class {} {{", name));
+				// create _toInflightType function that just requires the generated struct file
+				code.open(format!("static _toInflightType() {{"));
+				code.line(format!(
+					"return fromInline(`require(\"{}\")()`);",
+					struct_filename(&name.name)
+				));
+				code.close("}");
+
+				// Create validate function to check each struct field and return an array of errors
+				let obj = "obj".to_string();
+				code.open(format!("static validate({}) {{", &obj));
+				let invalid_errors = "invalidErrors".to_string();
+				code.line(format!("let {} = [];", &invalid_errors));
+				code.open(format!("if ({} === undefined) {{ ", &obj));
+				code.line("return [`obj is undefined`]");
+				code.close("}");
+				for field in fields {
+					if let Some(js_type) = self.jsify_type(&field.member_type.kind) {
+						let field_fqn = format!("{}.{}", &obj, &field.name.name);
+						let field_path = format!("{}.{}", name.name, &field.name.name);
+
+						code.open(format!(
+							"if ({}) {{",
+							self.jsify_struct_field_validation_check(&field.member_type.kind, &field_fqn)
+						));
+						code.line(format!(
+							"{}.push(`field: \"{}\" expected type of: \"{}\" but got type: ${{typeof {}}}`)",
+							&invalid_errors, &field_path, &js_type, &field_fqn
+						));
+						code.close("}");
+
+						// additional checks based on type of field
+						match &field.member_type.kind {
+							TypeAnnotationKind::Array(t) => {
+								code.open(format!("for(let i = 0; i < {}.length; i++) {{", &field_fqn));
+								code.open(format!(
+									"if({}) {{",
+									self.jsify_struct_field_validation_check(&t.kind, &format!("{}[i]", field_fqn))
+								));
+								code.line(format!(
+									"{}.push(`field: \"{}\" expected type of: \"{}\"`)",
+									&invalid_errors,
+									&field_path,
+									self.jsify_type(&field.member_type.kind).unwrap()
+								));
+								code.close("}");
+								code.close("}");
+							}
+							_ => {}
+						}
+					}
+				}
+
+				code.open(format!("if({}.length > 0) {{", &invalid_errors));
+				code.line(format!("console.log(`- ${{{}.join(\"\\n-\")}}`);", &invalid_errors));
+				code.close("}");
+				code.line(format!("return {}", &invalid_errors));
+				code.close("}");
+
+				// close class
+				code.close("}");
+				code.line(format!("return {};", name));
+				code.close("};");
+
+				// Emits struct class file
+				self.emit_struct_file(name, code, ctx);
+
+				// Reset the code maker for code to be inserted in preflight.js
+				code = CodeMaker::default();
+				code.line(format!(
+					"const {} = require(\"{}\")($stdlib.core.NodeJsCode.fromInline)",
```

Comment:
Perhaps just return a string and call fromInline here instead of passing in the function?

Reply:
hmm, so my intent was to reduce the boiler plating in the preflight.js file for what I am hoping is a readability improvement 🤞. Is there a concern with passing `fromInline`? I could totally be naive here